packages/opentelemetry-exporter-collector/src/transform.ts

```diff
@@ -202,11 +203,10 @@ export function toCollectorExportTraceServiceRequest<
   collectorExporterBase: CollectorExporterBase<T>,
   name = ''
 ): opentelemetryProto.collector.trace.v1.ExportTraceServiceRequest {
-  const spansToBeSent: opentelemetryProto.trace.v1.Span[] = spans.map(span =>
-    toCollectorSpan(span)
-  );
-  const resource: Resource =
-    spans.length > 0 ? spans[0].resource : Resource.empty();
+  const groupedSpans: Map<
+    Resource,
+    Map<core.InstrumentationLibrary, ReadableSpan[]>
+  > = groupSpans(spans);
```

Comment:
I think a more explicit (maybe even obnoxious) name might be a good idea here as "group" is fairly general. Something like `groupSpansByResourceAndLibrary`